pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -166,27 +164,6 @@ func validateServicePrincipalProfile(ctx context.Context, log *logrus.Entry, env
 	return refreshable.NewAuthorizer(token), nil
 }
 
-func (dv *openShiftClusterDynamicValidator) validateVnetPermissions(ctx context.Context, authorizer refreshable.Authorizer, client authorization.PermissionsClient, vnetID string, vnetr *azure.Resource, code, typ string) error {
```

Comment:
WDYT of keeping this function but move its content to the new `validate.ValidateVnetPermissions()` exported function? Then update this function to call `validate.ValidateVnetPermissions()`. That way, we can minimize changes to the dynamic validator and still ensure the new checker has an exported function to call?